### PR TITLE
Fix specs for `QueryLogsTest#test_sql_commenter_format`.

### LINKED ISSUE #52400: `QueryLogsTest#test_sql_commenter_format` fails
Managed to reproduce CI failure at https://buildkite.com/rails/rails/builds/109345#0190dd1d-ad92-4e8e-b1ba-4a66a8cea62c/1763-1769

### Steps to reproduce

```ruby
git clone https://github.com/rails/rails
cd rails/activerecord
bundle install
ARCONN=mysql2 bin/test test/cases/query_logs_test.rb -n test_sql_commenter_format
```

### Expected behavior
It should pass.

### Actual behavior
```ruby
$ ARCONN=mysql2 bin/test test/cases/query_logs_test.rb -n test_sql_commenter_format
Using mysql2
Run options: -n test_sql_commenter_format --seed 3251

# Running:

F

Failure:
QueryLogsTest#test_sql_commenter_format [test/cases/query_logs_test.rb:189]:
1 or more queries expected, but none were executed.
Queries:
SELECT `dashboards`.* FROM `dashboards` ORDER BY `dashboards`.`dashboard_id` ASC LIMIT 1.
Expected 0 to be >= 1.


bin/test test/cases/query_logs_test.rb:186



Finished in 0.041840s, 23.9008 runs/s, 47.8016 assertions/s.
1 runs, 2 assertions, 1 failures, 0 errors, 0 skips
$
```

### System configuration
**Rails version**: main branch

**Ruby version**: ruby 3.3.4 (2024-07-09 revision be1089c8ec) [x86_64-linux]

### PR DESCRIPTION
**Pull request by ruyrocha**
Fixes #52400

### Motivation / Background

This Pull Request has been created because of failing isolated test for `QueryLogsTest#test_sql_commenter_format` on main branch.

### Detail

This Pull Request changes [REPLACE ME]

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: activerecord/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+*   Fix specs for `QueryLogsTest#test_sql_commenter_format`.
```
Comment (Earlopain) [inline activerecord/CHANGELOG.md:1]: This has no user impact and doesn't require a changelog entry

**Review by skipkayhil — APPROVED** [review]

**Comment by jhawthorn** [comment]
Thank you!

**Comment by jhawthorn** [comment]
In a lot of ways, it's curious that this ever passed. It would be great if we could figure out what state is leaking that allows this test to _sometimes_ pass.

**Comment by yahonda** [comment]
As far as I understand this test had not passed since https://github.com/rails/rails/pull/52392

**Comment by jhawthorn** [comment]
> As far as I understand this test had not passed since #52392

Yep! A bisect confirms that. I still like the explicitness this PR introduced to the one test but I've opened #52410 which should turn the previous leaky state into an error.

**Pull request merged**